internal/controller/managedcluster_controller.go

```diff
@@ -993,6 +1000,33 @@ func setIdentityHelmValues(values *apiextensionsv1.JSON, idRef *corev1.ObjectRef
 	}, nil
 }
 
+func (r *ManagedClusterReconciler) setAvailableUpgrades(ctx context.Context, managedCluster *hmc.ManagedCluster, template *hmc.ClusterTemplate) error {
+	if template == nil || template.Labels[hmc.HMCManagedLabelKey] != hmc.HMCManagedLabelValue {
```

Comment:
I don't think we need to rely on these labels. And therefore I don't think we need the distinction between an empty list and nil here. managecluster-controller should not be aware of any subsequent validations and by default we should only allow upgrades within the chains. We might want to disable this validation on global or per-namespace basis though